crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs

```diff
@@ -51,15 +51,46 @@ pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> TokenStream {
             }
         });
 
+    #[cfg(feature = "documentation")]
+    let field_generator = {
+        let docs = reflect_struct
+            .active_fields()
+            .map(|field| quote::ToTokens::to_token_stream(&field.doc));
+        quote! {
+            #(#bevy_reflect_path::NamedField::new::<#field_types>(#field_names).with_doc(#docs) ,)*
+        }
+    };
+
+    #[cfg(not(feature = "documentation"))]
+    let field_generator = {
+        quote! {
+            #(#bevy_reflect_path::NamedField::new::<#field_types>(#field_names) ,)*
+        }
+    };
```

Comment:
I'm not sure of a better way of doing this unfortunately, as the `cfg!` macro doesn't strip code, it just returns a boolean.